contracts/Governance/TimelockV8.sol

```diff
@@ -0,0 +1,226 @@
+        bytes32 indexed txHash,
+        address indexed target,
+        uint256 value,
+        string signature,
+        bytes data,
+        uint256 eta
+    );
+
+    /// @notice Required period to execute a proposal transaction
+    uint256 private constant DEFAULT_GRACE_PERIOD = 14 days;
+
+    /// @notice Minimum amount of time a proposal transaction must be queued
+    uint256 private constant DEFAULT_MINIMUM_DELAY = 1 hours;
+
+    /// @notice Maximum amount of time a proposal transaction must be queued
+    uint256 private constant DEFAULT_MAXIMUM_DELAY = 30 days;
+
+    /// @notice Timelock admin authorized to queue and execute transactions
+    address public admin;
+
+    /// @notice Account proposed as the next admin
+    address public pendingAdmin;
+
+    /// @notice Period for a proposal transaction to be queued
+    uint256 public delay;
+
+    /// @notice Mapping of queued transactions
+    mapping(bytes32 => bool) public queuedTransactions;
+
+    constructor(address admin_, uint256 delay_) {
+        require(delay_ >= getMinimumDelay(), "Timelock::constructor: Delay must exceed minimum delay.");
+        require(delay_ <= getMaximumDelay(), "Timelock::setDelay: Delay must not exceed maximum delay.");
+
+        admin = admin_;
+        delay = delay_;
+    }
+
+    fallback() external payable {}
+
+    /**
+     * @notice Setter for the transaction queue delay
+     * @param delay_ The new delay period for the transaction queue
+     */
+    function setDelay(uint256 delay_) public {
+        require(msg.sender == address(this), "Timelock::setDelay: Call must come from Timelock.");
+        require(delay_ >= getMinimumDelay(), "Timelock::setDelay: Delay must exceed minimum delay.");
+        require(delay_ <= getMaximumDelay(), "Timelock::setDelay: Delay must not exceed maximum delay.");
+        delay = delay_;
+
+        emit NewDelay(delay);
+    }
+
+    function getGracePeriod() public view virtual returns (uint256) {
+        return DEFAULT_GRACE_PERIOD;
+    }
+
+    function getMinimumDelay() public view virtual returns (uint256) {
+        return DEFAULT_MINIMUM_DELAY;
+    }
+
+    function getMaximumDelay() public view virtual returns (uint256) {
+        return DEFAULT_MAXIMUM_DELAY;
+    }
+
+    /**
+     * @notice Method for accepting a proposed admin
+     */
+    function acceptAdmin() public {
+        require(msg.sender == pendingAdmin, "Timelock::acceptAdmin: Call must come from pendingAdmin.");
+        admin = msg.sender;
+        pendingAdmin = address(0);
+
+        emit NewAdmin(admin);
+    }
+
+    /**
+     * @notice Method to propose a new admin authorized to call timelock functions. This should be the Governor Contract
+     * @param pendingAdmin_ Address of the proposed admin
+     */
+    function setPendingAdmin(address pendingAdmin_) public {
+        require(msg.sender == address(this), "Timelock::setPendingAdmin: Call must come from Timelock.");
+        pendingAdmin = pendingAdmin_;
```

Comment:
Sanity check for zero address

Reply:
We could use https://github.com/VenusProtocol/solidity-utilities/blob/develop/contracts/validators.sol